public/test-cartridges/course-with-associated-content-assignments/imsmanifest.xml

```diff
@@ -90,6 +93,13 @@
       <file href="i9dede821e375f4888540a2095824f51e/assessment_meta.xml"/>
       <file href="non_cc_assessments/i9dede821e375f4888540a2095824f51e.xml.qti"/>
     </resource>
+    <resource identifier="e15f4285902a0458884f573e128eded9i" type="imsqti_xmlv1p2/imscc_xmlv1p1/assessment">
+      <dependency identifierref="icc89681ea9ad845e69a533543c3b1034"/>
```

Comment:
In the logic that adds the missing `file` element, can we change the logic to be based on the dependency instead of guessing a file from the `resource`' identifier?

Like `if no file` > `check dependency and grab its identifierref` > `find the dependency resource by identifierref` > `check files for qti XML` > `if found, copy the file element to the original node` > `otherwise do nothing`.

With this, we can reapply the filter at the end of the logic (that was based on the existing file), so it will be safer and more similar to the earlier behavior.

Reply:
I don't believe `assessment_qti.xml` is ever used. in the CC Viewer... other than to extract the identifier from the path. Also, looking at a `canvas_cc` generated manifest, I don't think we can safely get back to the identifier from the dependency node. 
```xml
  <item identifier="111111-module-item" identifierref="111111">
    <title>Some Title 1</title>
  </item>

  <resources>
    <resource type="imsqti_xmlv1p2/imscc_xmlv1p1/assessment" identifier="111111">
      <file href="111111/assessment_qti.xml"/> <!-- missing -->
      <dependency identifierref="111111_meta"/>
    </resource>
    <resource href="111111/assessment_meta.xml" identifier="111111_meta"
              type="associatedcontent/imscc_xmlv1p1/learning-application-resource">
      <file href="111111/assessment_meta.xml"/>
      <file href="non_cc_assessments/111111.xml.qti"/>
    </resource>
```

Reply:
`111111/assessment_qti.xml` will be an existing file at the end?
From your example in the code, it looks to me that `${indentifier}/assessment_qti.xml` does not exist.

Reply:
I mean, the current logic will just check whether the node has or doesn't have a file node, it will add it.
I'm not sure that it is always a good scenario. What I suggest is just one more layer to check if we need to add that file node, or not.

Reply:
OK, I can work on that. We need better test examples based on the output from canvas_cc gem and Canvas. The identified is ripped out of the file node href [here](https://github.com/instructure/common-cartridge-viewer/blob/master/src/utils.js#L236).

Reply:
@woodie I've also tried it and got some weird results. Let's go with a smaller check if you can't figure out a proper connection between the dependency and the files.

```
  const assessmentResources = resources
    .filter(is(resourceTypes.ASSESSMENT_CONTENT))
    .map(node => {
      if (!node.querySelector("file") && node.querySelector("dependency")) {
        // add missing file node for canvas_cc generated resources
        const fileNode = document.createElement("file");
        const href = node.getAttribute("identifier") + "/assessment_qti.xml";
        fileNode.setAttribute("href", href);
        node.appendChild(fileNode);
      }
      return node;
    }).filter(node => node.querySelector("file"));
```

Added `node.querySelector("dependency")` and reintroduced the last filter.

Reply:
By the way, [here](https://github.com/instructure/canvas_cc/blob/master/lib/canvas_cc/canvas_cc/models/assessment.rb#L30) is where `canvas_cc` creates the resource dependencies with `["#{@identifier}_meta"]`.